kiwiapp-public/src/components/common/MedicationTable.tsx

```diff
@@ -0,0 +1,134 @@
+    () => [
+      {
+        accessorKey: "name",
+        header: () => "Medication Name",
+        cell: ({ getValue }) => <span className="text-left">{getValue<string>()}</span>,
+      },
+      {
+        accessorKey: "dosage",
+        header: () => "Dosage",
+        cell: ({ getValue }) => <span className="text-center">{getValue<string>()}</span>,
+      },
+      {
+        accessorKey: "quantity",
+        header: () => "Quantity",
+        cell: ({ getValue }) => <span className="text-center">{getValue<number>()}</span>,
+        size: 70,
+      },
+      {
+        accessorKey: "daysSupply",
+        header: () => "Days Supply",
+        cell: ({ getValue }) => <span className="text-center">{getValue<number>()}</span>,
+      },
+      {
+        id: "actions",
+        header: () => "",
+        cell: ({ row }) => (
+          <Button
+            className="h-10 w-10 rounded-full bg-red-500 p-0 text-white hover:bg-red-700"
+            onClick={handleDelete(row.index)}
+          >
+            <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+              <g>
+                <g>
+                  <path
+                    d="M16.1032 7.89062C16.1032 7.89062 15.6507 13.5031 15.3882 15.8673C15.2632 16.9965 14.5657 17.6581 13.4232 17.679C11.249 17.7181 9.07234 17.7206 6.89901 17.6748C5.79984 17.6523 5.11401 16.9823 4.99151 15.8731C4.72734 13.4881 4.27734 7.89062 4.27734 7.89062"
+                    stroke="white"
+                    strokeWidth="1.5"
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                  />
+                  <path
+                    d="M17.2567 5.19987H3.125"
+                    stroke="white"
+                    strokeWidth="1.5"
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                  />
+                  <path
+                    d="M14.5335 5.19998C13.8793 5.19998 13.316 4.73748 13.1877 4.09665L12.9852 3.08331C12.8602 2.61581 12.4368 2.29248 11.9543 2.29248H8.42682C7.94432 2.29248 7.52099 2.61581 7.39599 3.08331L7.19349 4.09665C7.06516 4.73748 6.50182 5.19998 5.84766 5.19998"
+                    stroke="white"
+                    strokeWidth="1.5"
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                  />
+                </g>
+              </g>
+            </svg>
+          </Button>
+        ),
+      },
+    ],
+    [handleDelete],
+  );
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  return (
+    <div className="w-full">
+      <table className="w-full">
+        <thead>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <tr
+              key={headerGroup.id}
+              className="grid grid-cols-[2.1fr_1fr_1.3fr_1.4fr_40px] md:grid-cols-[1.6fr_1fr_2.3fr_0.9fr_60px]"
```

Comment:
Avoid customized grid column values. You should define the number of columns here, and then use `col-span` to define how many grid columns that each display column will be using.